src/app/clusters/service-area-server/service-area-server.cpp

```diff
@@ -0,0 +1,1504 @@
+        {
+            mSelectedLocations.push_back(newSelectLocations[locationIndex]);
+        }
+
+        NotifySelectedLocationsChanged();
+    }
+
+
+exit:
+    response.status = locationStatus;
+    if (locationStatusText.length() > 0)
+    {
+        response.statusText = chip::CharSpan::fromCharString( locationStatusText.c_str());
+    }
+
+    ctx.mCommandHandler.AddStatus(ctx.mRequestPath, cmdStatus);
+    ctx.mCommandHandler.AddResponse(ctx.mRequestPath, response);
+}
+
+
+bool Instance::AreSetSelectLocationsParamsValid(const std::vector<uint32_t> & selectLocations, 
+                                            SelectLocationsStatus & locationStatus, std::string & locationStatusText)
+{
+    bool ret_val = false;
+    std::map<uint32_t, uint32_t> testMap;
+
+    for (size_t entry = 0; entry < selectLocations.size(); ++entry)
+    {
+        // If at least one entry on the NewLocations field doesn't match the LocationID field of any entry of the SupportedLocations list,
+        // the SelectLocationsResponse command's Status field SHALL indicate UnsupportedLocation.
+        // If the Status field is set to UnsupportedLocation, the StatusText field SHALL be an empty string.
+        VerifyOrExit(IsSupportedLocation(selectLocations[entry]), 
+                            locationStatus = SelectLocationsStatus::kUnsupportedLocation;
+                            ChipLogError(Zcl, "AreSetSelectLocationsParamsValid - unsupported location %u", selectLocations[entry]) );
+
+        // If this field contains any duplicated entries,
+        // the SelectLocationsResponse command's Status field SHALL indicate DuplicatedLocations.
+        // If the Status field is set to DuplicatedLocations, the StatusText field SHALL be an empty string.
+        VerifyOrExit(testMap.emplace(entry, entry).second, 
+                        locationStatus = SelectLocationsStatus::kDuplicatedLocations;
+                        ChipLogError(Zcl, "AreSetSelectLocationsParamsValid - duplicated location %u", selectLocations[entry]));
+    }
+
+
+    // all checks passed,
+    ret_val = true;
+    locationStatus = SelectLocationsStatus::kSuccess;
+    // If the Status field is set to Success, the StatusText field is optional.
+
+
+exit:
+    return ret_val;
+}
+
+
+void Instance::HandleSkipCurrentCmd(HandlerContext & ctx)
+{
+    ChipLogDetail(Zcl, "Location Location: HandleSkipCurrent");
+
+    SkipCurrentStatus skipStatus = SkipCurrentStatus::kSuccess;
+    std::string       skipStatusText; 
+
+    Status   cmdStatus = Status::Success;
+
+    // On receipt of this command the device SHALL respond with a SkipCurrentLocationResponse command. 
+    Commands::SkipCurrentResponse::Type response;
+
+    // InvalidLocationList | The SelectedLocations attribute is null.
+    // If the Status field is set to InvalidLocationList, the StatusText field SHALL be an empty string.
+    VerifyOrExit((mSelectedLocations.size() == 0), 
```

Comment:
Internally, do we differentiate between an empty SelectedLocations attribute and a null SelectedLocations attribute?

Reply:
The assumption is that an empty mSelectedLocations list indicates a null Selected Locations attribute, similar to all the other list attributes.

It's not explicitly called out for SelectedLocations in the spec with a constraint like the other lists, it would be consistent with how the Progress attribute is defined to have a constraint of 1 to 255. (Both must have a subset of Supported Locations without duplicates, so the working  limit is dynamic and equal to Supported Locations size() -  and likely less than that if multiple maps are defined.)

The requirements explicitly state how to handle null Selected Locations and how to handle a null Selected Locations with values. They say nothing about what to do with an empty list.

Reply:
Right, that's why I was asking. If SelectedLocations has one entry and we delete the associated map, or edit the location's info, that list becomes empty. The spec currently says that "If the SelectedLocations attribute list becomes empty, the server SHALL stop operating". The reason is that a null, rather than empty, SelectedLocations means "operate anywhere possible".

We could change the spec and say that if all entries of SelectedLocations are removed, the attribute is null-ed and the device stops operating. 

Let me know your thoughts (both you and @hicklin, please).

Reply:
I'd favor changing the spec - constraint 1 to 255, revised language as you suggest.

I don't see how an empty list in addition to a null list adds any value.

Reply:
I think I see it the other way. Can we work with just 0 and no null? If so, this would be simpler to deal with.

Reply:
I agree that non-nullable elements are easier to deal with regarding syntax etc. and was annoyed at all of the IsNull() and Value() syntax nullable elements pull in.

Unfortunately, I looked at how other clusters deal with this, and general usage seems to prefer nullable arrays.
The convention is - if there is no usable data, should be NULL. ?

Reply:
OK. So we're going to update the spec to clarify that `0` is not a valid number of SupportedMaps. In the SDK and user applications, if we have 0 elements it means `Null`.